src/Features/Core/Portable/AddImport/References/Reference.cs

```diff
@@ -107,11 +107,15 @@ public override int GetHashCode()
 
                 var root = contextNode.SyntaxTree.GetRoot(cancellationToken);
                 root = root.ReplaceToken(identifier, newIdentifier.WithAdditionalAnnotations(annotation));
-                document = document.WithSyntaxRoot(root);
-                contextNode = root.GetAnnotatedTokens(annotation).First().Parent;
```

Comment:
unfortunately, when you make a document out of a new root compilation unit, the document's root won't be that same object.  Do you need to go reacquire the root after making the new document.